rclpy/src/rclpy/wait_set.hpp

```diff
@@ -30,149 +31,151 @@ namespace py = pybind11;
- * \param[in] client a client to add to the wait set
- * \return Index in waitset entity was added at
- */
-size_t
-wait_set_add_client(const py::capsule pywait_set, const Client & client);
-
-/// Add a timer to the wait set structure
-/**
- * Raises RCLError if any lower level error occurs
- *
- * \param[in] pywait_set Capsule pointing to the wait set structure
- * \param[in] timer a timer to add to the wait set
- * \return Index in waitset entity was added at
- */
-size_t
-wait_set_add_timer(const py::capsule pywait_set, const Timer & timer);
-
-/// Add an event to the wait set structure
-/**
- * Raises RCLError if any lower level error occurs
- *
- * \param[in] pywait_set Capsule pointing to the wait set structure
- * \param[in] event a QoSEvent to add to the wait set
- * \return Index in waitset entity was added at
- */
-size_t
-wait_set_add_event(const py::capsule pywait_set, const QoSEvent & event);
-
-/// Check if an entity in the wait set is ready by its index
-/**
- * This must be called after waiting on the wait set.
- * Raises RuntimeError if the entity type is unknown
- * Raises IndexError if the given index is beyond the number of entities in the set
- *
- * \param[in] entity_type string defining the entity ["subscription, client, service"]
- * \param[in] pywait_set Capsule pointing to the wait set structure
- * \param[in] index location in the wait set of the entity to check
- * \return True if the entity at the index in the wait set is not NULL
- */
-bool
-wait_set_is_ready(const std::string & entity_type, py::capsule pywait_set, size_t index);
-
-/// Get list of non-null entities in wait set
-/**
- * Raises ValueError if pywait_set is not a wait set capsule
- * Raises RuntimeError if the entity type is not known
- *
- * \param[in] entity_type string defining the entity ["subscription, client, service"]
- * \param[in] pywait_set Capsule pointing to the wait set structure
- * \return List of wait set entities pointers ready for take
- */
-py::list
-get_ready_entities(const std::string & entity_type, py::capsule pywait_set);
-
-/// Wait until timeout is reached or event happened
-/**
- * Raises ValueError if pywait_set is not a wait set capsule
- * Raises RCLError if there was an error while waiting
- *
- * This function will wait for an event to happen or for the timeout to expire.
- * A negative timeout means wait forever, a timeout of 0 means no wait
- * \param[in] pywait_set Capsule pointing to the wait set structure
- * \param[in] timeout optional time to wait before waking up (in nanoseconds)
- */
-void
-wait(py::capsule pywait_set, int64_t timeout);
+class WaitSet : public Destroyable, public std::enable_shared_from_this<WaitSet>
+{
+public:
+  /// Initialize a wait set
+  WaitSet();
+
+  /// Initialize a wait set
+  /**
+   * Raises RCLError if the wait set could not be initialized
+   *
+   * \param[in] node_name string name of the node to be created
+   * \param[in] number_of_subscriptions int
```

Comment:
@ahcorde nit: 
```suggestion
   * \param[in] number_of_subscriptions a positive number or zero
```
Same elsewhere.